search/search.go

```diff
@@ -0,0 +1,36 @@
+package search
+
+import (
+	"encoding/json"
+)
+
+type Engine interface {
+	SuggestProjects(index, q string) (map[string]json.RawMessage, error)
+}
+
+type Searcher struct {
```

Comment:
Why do you need both Engine and the wrapping Searcher? This can just be 1  struct.

Reply:
As long as the engines have the same interface, they can be swapped, the facade is superfluous.

Reply:
Well [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). I'll remove this.

Reply:
Done!